nntrainer/layers/permute_layer.cpp

```diff
@@ -76,10 +76,18 @@ int PermuteLayer::initialize(Manager &manager) {
   return ML_ERROR_NONE;
 }
 
-void PermuteLayer::forwarding(bool training) { /** NYI */
+void PermuteLayer::forwarding(bool training) {
+  auto &input_ = net_input[0]->getVariableRef();
+  auto &hidden_ = net_hidden[0]->getVariableRef();
+
+  input_.transpose(direction_str, hidden_);
 }
 
-void PermuteLayer::calcDerivative() { /** NYI */
+void PermuteLayer::calcDerivative() {
+  auto &input_grad = net_input[0]->getGradientRef();
+  auto &hidden_grad = net_hidden[0]->getVariableRef();
```

Comment:
In the backwarding I think the hidden_grad should be net_hidden[0]->getGradientRef() rather than getVariableRef(), isn't it?

Reply:
Nice catch! this should be getGradientRef(), thanks :)

Reply:
@zhoonit Please update this before merge.